idtype/idtype.go

```diff
@@ -153,12 +164,39 @@ func getDefn(t Type) definition {
 // Any types used *must* be registered. Ideally, call this from your package's
 // init function.
 //
+// The third argument of register are the options. It currently only support one option, setting the plural version
+// of the name. If omitted, the name will have an s appended to generate its plural version.
+//
 // Register panics if it is called twice with the same Type.
-func Register(typ Type, mutable bool, name string) {
+func Register(typ Type, mutable bool, name string, options ...interface{}) {
```

Comment:
Ideally we would have an Option type like the functional option link I shared instead of the opaque `interface{}`. But this works and is extensible.